### DISCUSSION TITLE
Using own account on forking together with other signers

### DISCUSSION THREAD

**Discussion by GorniyGor** [NomicFoundation/hardhat#3562] [Troubleshooting] [answered]
I use forking, I need to be able signMessage (not only txn) but I need couple random accs in test as well.

My attempts:
1) use regular _getSigners_ for support accs + use _impersonate_ for a certain (mine) -> do work with signTransaction, but don't with signMessage
2) use regular _getSigners_ for support accs + use _Wallet_ for a certain (mine, I have private key) -> I haven't figured out how to put proper Provider in Wallet (provider of my local forking)
3) use regular _getSigners_ for support accs + set a certain _account_ into `harghat.config.js` (`networks/hardhat/accounts`) -> do work with signMessage, but getSigners return only 1 acc - owner (from config), and returned no additional random signers

Is there a way to solve my problem? (use hardhat + ethers)

**Comment by fvictorio** [comment, marked as answer]
I think the second option is the way to go. The way to use the right provider is this:

```js
const wallet = new ethers.Wallet(privateKey, ethers.provider)
```

**Reply by GorniyGor** [reply]
It doesn't work for `forking` -> I receive error that `provider` is not defined (smth like that)

**Reply by fvictorio** [reply]
Are you using typescript? If so, are you sure the `ethers` object is imported from hardhat, and not from ethers? That is, it should look like this:

```typescript
import { ethers } from "hardhat"
```

and not

```typescript
import { ethers } from "ethers"
```

**Reply by GorniyGor** [reply]
I use JS, and import from hardhat, yes.

**Reply by GorniyGor** [reply]
@fvictorio oh, sorry, looks like I had used just `provider`, not `ethers.provider`.
It works, thanks a lot!
